.github/workflows/publish-providers.yml

```diff
@@ -13,12 +21,171 @@ on:
+          cache: yarn
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Build package
+        run: |
+          cd packages/${{ matrix.package }}
+          yarn build
+
+      - name: Conventional Changelog Action
+        id: changelog
+        uses: TriPSs/conventional-changelog-action@3a392e9aa44a72686b0fc13259a90d287dd0877c
+        with:
+          git-user-name: ${{ steps.github-user-info.outputs.username }}
+          git-user-email: ${{ steps.github-user-info.outputs.email }}
+          github-token: ${{ secrets.SVC_CLI_BOT_GITHUB_TOKEN }}
+          tag-prefix: "${{ matrix.package }}@"
+          release-count: "0"
+          skip-on-empty: ${{ github.event_name == 'push' || inputs.skip-on-empty }}
+          git-path: "packages/${{ matrix.package }}"
+          version-file: "packages/${{ matrix.package }}/package.json"
+          output-file: "packages/${{ matrix.package }}/CHANGELOG.md"
+
+      - name: Create Github Release
+        id: release
+        uses: ncipollo/release-action@2c591bcc8ecdcd2db72b97d6147f871fcd833ba5
+        if: ${{ steps.changelog.outputs.skipped == 'false' }}
+        with:
+          name: "${{ matrix.package }}@${{ steps.changelog.outputs.version }}"
+          tag: "${{ matrix.package }}@${{ steps.changelog.outputs.version }}"
+          commit: ${{ github.sha }}
+          body: |
+            ## Changes in ${{ matrix.package }}
+            
+            ${{ steps.changelog.outputs.clean_changelog }}
+          token: ${{ secrets.SVC_CLI_BOT_GITHUB_TOKEN }}
+          skipIfReleaseExists: true
+
+      - name: Publish to npm
+        if: ${{ steps.changelog.outputs.skipped == 'false' && steps.release.outputs.id != '' }}
+        run: |
+          cd packages/${{ matrix.package }}
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" > ~/.npmrc
+          npm publish --access public
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Update main server dependency
+        if: ${{ steps.changelog.outputs.skipped == 'false' && steps.release.outputs.id != '' }}
+        run: |
+          # Get the published version
+          PUBLISHED_VERSION="${{ steps.changelog.outputs.version }}"
+          PACKAGE_NAME="@salesforce/${{ matrix.package }}"
+          
+          echo "Updating $PACKAGE_NAME to version $PUBLISHED_VERSION in main MCP server"
+          
+          # Update the dependency in main server's package.json using jq
+          cd packages/mcp
+          jq --arg pkg "$PACKAGE_NAME" --arg ver "^$PUBLISHED_VERSION" \
+            '.dependencies[$pkg] = $ver' package.json > package.json.tmp
+          mv package.json.tmp package.json
+          
+          # Go back to root and update yarn.lock
+          cd ../../
+          # TODO(cristian): need to nuke all `node_modules` to cleanup some dep, running `yarn install` 2 times fails at the second run.
+          git clean -fdx
+          yarn install
```

Comment:
Note for the future: We may want to add a `npx yarn-deduplicate && yarn install` here